s3-ferry/file_api.py

```diff
@@ -0,0 +1,204 @@
+from fastapi import FastAPI, File, UploadFile, HTTPException, Form, Request
+from fastapi.responses import FileResponse, JSONResponse
+import os
+import requests
+from file_converter import FileConverter
+import json
+from pydantic import BaseModel
+
+app = FastAPI()
+
+UPLOAD_DIRECTORY = os.getenv("UPLOAD_DIRECTORY", "/shared")
+RUUTER_PRIVATE_URL = os.getenv("RUUTER_PRIVATE_URL")
+S3_FERRY_URL = os.getenv("S3_FERRY_URL")
+
+class ExportFile(BaseModel):
+    dg_id: int
+    version: str
+    export_type: str
+
+if not os.path.exists(UPLOAD_DIRECTORY):
+    os.makedirs(UPLOAD_DIRECTORY)
+
+def get_ruuter_private_url():
+    return os.getenv("RUUTER_PRIVATE_URL")
+
+async def authenticate_user(request: Request):
+    cookie = request.cookies.get("customJwtCookie")
+    if not cookie:
+        raise HTTPException(status_code=401, detail="No cookie found in the request")
+
+    url = f"{RUUTER_PRIVATE_URL}/auth/jwt/userinfo"
+    headers = {
+        'cookie': f'customJwtCookie={cookie}'
+    }
+
+    response = requests.get(url, headers=headers)
+    if response.status_code != 200:
+        raise HTTPException(status_code=response.status_code, detail="Authentication failed")
+
+@app.post("/datasetgroup/data/import")
+async def upload_and_copy(request: Request, dg_id: int = Form(...), import_type: str = Form(...), data_file: UploadFile = File(...)):
+    await authenticate_user(request)
+    if import_type not in ["major","minor"]:
+        raise HTTPException(
+            status_code=500,
+            detail={
+                "upload_status": 500,
+                "operation_successful": False,
+                "saved_file_path": None,
+                "reason" : "import_type should be either minor or major."
+            }
+        )
+
+    file_location = os.path.join(UPLOAD_DIRECTORY, data_file.filename)
+    file_name = data_file.filename
+    with open(file_location, "wb") as f:
+        f.write(data_file.file.read())
+
+    file_converter = FileConverter()
+    success, converted_data = file_converter.convert_to_json(file_location)
+    if not success:
+        raise HTTPException(
+            status_code=500,
+            detail={
+                "upload_status": 500,
+                "operation_successful": False,
+                "saved_file_path": None,
+                "reason" : "Json file convert failed."
+            }
+        )
+    
+    json_local_file_path = file_location.replace('.yaml', '.json').replace('.yml', '.json').replace('.xlsx', ".json")
+    with open(json_local_file_path, 'w') as json_file:
+        json.dump(converted_data, json_file, indent=4)
+
+    if import_type == "minor":
+        save_location = f"/dataset/{dg_id}/minor_update_temp/minor_update_.json"
+    elif import_type == "major":
+        save_location = f"/dataset/{dg_id}/primary_dataset/dataset_{dg_id}_aggregated.json"
```

Comment:
I think in our initial discussion we spoke that the /import endpoint which is ONLY used by the frontend will upload files ONLY for minor updates. In major update we have only schema updates so it will never call this endpoint. For patch updates the data update is passed to the validator and then preprocessor.py as a JSON payload, again the /import endpoint is not called. What's our reasoning behind this logic here @pamodaDilranga ?

Reply:
This is cleared and resolved